x/ccv/provider/migrations/v5/migrations.go

```diff
@@ -10,9 +10,11 @@ import (
 // This migration only takes already registered chains into account.
 // If a chain is in voting while the upgrade happens, this is not sufficient,
 // and a migration to rewrite the proposal is needed.
+//
+// TODO (PERMISSIONLESS): this migration needs to be fix or removed
```

Comment:
This needs to be fixed as the keeper methods changed. This means that nobody can run the v5 (or v6) migration without going first through v8. Thus, these migration require having legacy keeper methods.